src/promise-btn.directive.ts

```diff
@@ -195,26 +194,24 @@ export class PromiseBtnDirective implements OnDestroy, AfterContentInit {
   }
 
   /**
-   * Used to limit loading state to show only for the currently
-   * clicked button.
-   * @param {Object}btnEl
+   * Limit loading state to show only for the currently clicked button.
+   * Executed only if this.cfg.handleCurrentBtnOnly is set
    */
-  addHandlersForCurrentBtnOnlyIfSet(btnEl: HTMLElement) {
-    // handle current button only options via click
-    if (this.cfg.handleCurrentBtnOnly) {
-      btnEl.addEventListener(this.cfg.CLICK_EVENT, () => {
-        // return if something else than a promise is passed
-        if (!this.promise) {
-          return;
-        }
-
-        // due to some really weird reasons, we need a timeout
-        // to let the model still update when a button
-        // inside a form is disabled
-        setTimeout(() => {
-          this.initLoadingState(btnEl);
-        }, 1);
-      });
```

Comment:
I replaced addEventListener with HostListener as the [angular guide suggests](https://angular.io/guide/attribute-directives#respond-to-user-initiated-events). 

> Of course you could reach into the DOM with standard JavaScript and and attach event listeners manually. There are at least three problems with that approach:

> You have to write the listeners correctly.
The code must detach the listener when the directive is destroyed to avoid memory leaks.
Talking to DOM API directly isn't a best practice.

It also fixed a memory leak, as the event listener wasn't detached when directive is destroyed.